From 123bb3de31a26dc667d4e7340e19e0189fb209c2 Mon Sep 17 00:00:00 2001
From: Cyril Hrubis <chrubis@suse.cz>
Date: Mon, 21 Jan 2013 16:11:26 +0100
Subject: [PATCH] syscalls/mremap05: Fix build.

Make use of allredy existing
HAVE_MREMAP_FIXED macro from config.h.

Signed-off-by: Cyril Hrubis <chrubis@suse.cz>
---
 testcases/kernel/syscalls/mremap/mremap05.c |   12 ++++++++++++
 1 files changed, 12 insertions(+), 0 deletions(-)

diff --git a/testcases/kernel/syscalls/mremap/mremap05.c b/testcases/kernel/syscalls/mremap/mremap05.c
index e037be4..07ec49f 100644
--- a/testcases/kernel/syscalls/mremap/mremap05.c
+++ b/testcases/kernel/syscalls/mremap/mremap05.c
@@ -36,6 +36,7 @@
  */
 
 #define _GNU_SOURCE
+#include "config.h"
 #include <sys/mman.h>
 #include <errno.h>
 #include <unistd.h>
@@ -44,6 +45,8 @@
 
 char *TCID = "mremap05";
 
+#ifdef HAVE_MREMAP_FIXED
+
 struct test_case_t {
 	char *old_address;
 	char *new_address;
@@ -241,3 +244,12 @@ static void cleanup(void)
 {
 	TEST_CLEANUP;
 }
+
+#else
+
+int main(void)
+{
+	tst_brkm(TCONF, NULL, "MREMAP_FIXED not present in <sys/mman.h>");
+}
+
+#endif /* HAVE_MREMAP_FIXED */
-- 
1.7.8.6

